### ISSUE TITLE
Calling `window.EventSource` causes an error in the Bun engine.

### ISSUE THREAD

**Issue by jackens** [capricorn86/happy-dom#1085]
**Describe the bug**

Calling `window.EventSource` causes an error in the *Bun* engine.

**To Reproduce**

`bug.js`:

```js
import { GlobalRegistrator } from '@happy-dom/global-registrator'

GlobalRegistrator.register()

console.log(window.EventSource)
```

Executing the command:

```sh
bun run bug.js
```

produces the following error:

```log
1 | (function () {"use strict";
                       ^
TypeError: undefined is not an object or null
      at getEventSource (:1:20)
      at /path/to/bug.js:5:12
```

**Expected behavior**

`window.EventSource` works with *Bun*

**Device:**

- OS: macOS 13.4.1 (c) (22F770820d)
- JS Engine: Bun 1.0.2
- `@happy-dom/global-registrator`: 12.1.5

**Additional context**

In version 12.1.4 of `@happy-dom/global-registrator`, the error does not occur.

**Comment by croqaz** [comment]
I wanted to use happy-dom, I can confirm this happens like described, even with latest bun `1.0.6`.

**Comment by jackens** [comment]
Fixed in 493dd8045a71053d58ac05aec3578363645467be
